acceptancetests/update-jenkins-slave.bash

```diff
@@ -19,10 +19,26 @@ update_branch() {
 update_git_repo() {
     # Clone or pull a git repo.
     git_repo=$1
-    local_dir="$(echo $git_repo|sed -r 's/.*\/([^\/]*)\.git/\1/')"
+    branch=$2
+    local_dir=$3
     local_path="$HOME/$local_dir"
     if [[ -d $local_path ]]; then
-        (cd $local_path; git pull $git_repo)
+        # Clear any local changes
+        if [[ -d "$local_path/.bzr" ]]; then
+            # check for .bzr, if so, do fresh clone
+            echo "Found bzr repo, removing and cloning"
+            git clone $git_repo "$local_path-git"
+            rm -rf $local_path
+            mv "$local_path-git" "$local_path"
+        else
+            cd $local_path
```

Comment:
You change the working directory here but don't return to the original.
This might have the possiblity of breaking things down the line as where in a new directory now.